Fix compiler warnings.

There are couple of warnings to be seen during the build process and it should be quite easy to get rid of them:

```
4>------ Rebuild All started: Project: Stratis.Bitcoin, Configuration: Debug Any CPU ------
4>Features\MemoryPool\MempoolValidator.cs(554,4,554,6): warning CS0162: Unreachable code detected
4>Features\MemoryPool\MempoolOrphans.cs(192,11,192,22): warning CS0219: The variable 'nFetchFlags' is assigned but its value is never used
4>Features\Consensus\ConsensusStats.cs(15,31,15,36): warning CS0169: The field 'ConsensusStats.stack' is never used
4>Features\RPC\RPCRouteHandler.cs(20,38,20,54): warning CS0169: The field 'RPCRouteHandler.contollerMapping' is never used
4>Features\Miner\PosMinting.cs(48,25,48,38): warning CS0169: The field 'PosMinting.hashPrevBlock' is never used
```

```
5>------ Rebuild All started: Project: Stratis.Bitcoin.IntegrationTests, Configuration: Debug Any CPU ------
5>UtilTests.cs(39,6,39,48): warning CS4014: Because this call is not awaited, execution of the current method continues before the call is completed. Consider applying the 'await' operator to the result of the call.
5>MemoryPoolTests.cs(946,14,946,16): warning CS0649: Field 'TestMemPoolEntryHelper.lp' is never assigned to, and will always have its default value null
5>BlockStoreTests.cs(20,15,20,35): warning xUnit1013: Public method 'BlockRepositoryBench' on test class 'BlockStoreTests' should be marked as a Fact.
5>MinerTests.cs(567,15,567,42): warning xUnit1013: Public method 'MinerCreateNewBlockValidity' on test class 'MinerTests' should be marked as a Fact.
```

--- a/Stratis.Bitcoin.IntegrationTests/BlockStoreTests.cs
+++ b/Stratis.Bitcoin.IntegrationTests/BlockStoreTests.cs
@@ -15,9 +15,7 @@ namespace Stratis.Bitcoin.IntegrationTests
 {
     public class BlockStoreTests
     {
-
-		// [Fact]
-		public void BlockRepositoryBench()
+		private void BlockRepositoryBench()
 		{
 			using (var dir = TestDirectory.Create())
 			{

--- a/Stratis.Bitcoin.IntegrationTests/MinerTests.cs
+++ b/Stratis.Bitcoin.IntegrationTests/MinerTests.cs
@@ -563,7 +563,7 @@ namespace Stratis.Bitcoin.IntegrationTests
 		}
 
 		//NOTE: These tests rely on CreateNewBlock doing its own self-validation!
-		public void MinerCreateNewBlockValidity()
+		private void MinerCreateNewBlockValidity()
 		{
 			//TODO: fix this test
 			// orphan in mempool, template creation fails

--- a/Stratis.Bitcoin.IntegrationTests/UtilTests.cs
+++ b/Stratis.Bitcoin.IntegrationTests/UtilTests.cs
@@ -36,12 +36,13 @@ namespace Stratis.Bitcoin.IntegrationTests
 				{
 					collector.Add(1);
 					// push another exclusive task to the scheduler
-					session.WriteAsync(() => collector.Add(2));
+					Task exclusiveTask = session.WriteAsync(() => collector.Add(2));
 					// await a concurrent task, this will split the current method in two tasks
 					// the pushed exclusive task will processes before the await yields back control
 				    await session.ReadAsync(() =>  collector.Add(3));
 					collector.Add(4);
-				});
+                    await exclusiveTask;
+                });
 
 			});
 

--- a/Stratis.Bitcoin/Features/Consensus/ConsensusStats.cs
+++ b/Stratis.Bitcoin/Features/Consensus/ConsensusStats.cs
@@ -12,7 +12,6 @@ namespace Stratis.Bitcoin.Features.Consensus
 {
     public class ConsensusStats
     {
-        private CoinViewStack stack;
         private CachedCoinView cache;
         private DBreezeCoinView dbreeze;
         private CoinView bottom;

--- a/Stratis.Bitcoin/Features/Miner/PosMinting.cs
+++ b/Stratis.Bitcoin/Features/Miner/PosMinting.cs
@@ -44,7 +44,6 @@ namespace Stratis.Bitcoin.Features.Miner
 		private readonly PosConsensusValidator posConsensusValidator;
 	    private readonly ILogger logger;
 
-        private uint256 hashPrevBlock;
 		private Task mining;
 		private readonly long lastCoinStakeSearchTime;
 		private Money reserveBalance;

--- a/Stratis.Bitcoin/Features/RPC/RPCRouteHandler.cs
+++ b/Stratis.Bitcoin/Features/RPC/RPCRouteHandler.cs
@@ -17,8 +17,6 @@ namespace Stratis.Bitcoin.Features.RPC
 
 	public class RPCRouteHandler : IRPCRouteHandler
 	{
-		private Dictionary<string, string> contollerMapping;
-
 		private IRouter inner;
 		private IActionDescriptorCollectionProvider actionDescriptor;
 

